code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -461,3 +461,54 @@
 	//NC.mergeConnectedNetworksOnTurf()
 	last_piece = NC
 	return 1
+
+//Dunno where else to put this so shrug
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade
+	name = "Ripley MK-II Conversion Kit"
+	desc = "A pressurized canopy attachment kit for an Autonomous Power Loader Unit \"Ripley\" MK-I mecha, to convert it to the slower, but space-worthy MK-II design. This kit cannot be removed, once applied."
+	icon_state = "ripleyupgrade"
+
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade/can_attach(obj/mecha/working/ripley/M)
+	if(M.type != /obj/mecha/working/ripley)
+		to_chat(loc, "<span class='warning'>This conversion kit can only be applied to APLU MK-I models.</span>")
+		return 0
+	if(M.cargo.len)
+		to_chat(loc, "<span class='warning'>[M]'s cargo hold must be empty before this conversion kit can be applied.</span>")
+		return 0
+	if(!M.maint_access) //non-removable upgrade, so lets make sure the pilot or owner has their say.
+		to_chat(loc, "<span class='warning'>[M] must have maintenance protocols active in order to allow this conversion kit.</span>")
+		return 0
+	if(M.occupant) //We're actualy making a new mech and swapping things over, it might get weird if players are involved
+		to_chat(loc, "<span class='warning'>[M] must be unoccupied before this conversion kit can be applied.</span>")
+		return 0
+	return 1
+
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade/attach(obj/mecha/M)
+	var/obj/mecha/working/ripley/mkii/N = new /obj/mecha/working/ripley/mkii(get_turf(M),1)
+	if(!N)
+		return
+	qdel(N.cell)
+	N.cell = null
+	if (M.cell)
+		N.cell = M.cell
+		M.cell.forceMove(N)
+		M.cell = null
+	N.step_energy_drain = M.step_energy_drain //For the scanning module
```

Comment:
lol does it delete the stock parts on construction? I hate mech code...

Reply:
Imagine trying to figure this all out. It's a mess.